### ISSUE TITLE
Skipping entries with InvalidFormatException

### ISSUE THREAD

**Issue by pgodwin** [adamhathcock/sharpcompress#656]
Firstly, thanks for such an terrific library!

Quick question - is it possible to skip entries with an InvalidFormatException? 

I have some rar files (comicbook CBR files), that seem to have been corrupted. WinRar will skip over those files, but SharpCompress throws an `SharpCompress.Common.InvalidFormatException: 'Unknown Rar Header: 205'` exception. 

It seems to happen on the MoveNext() of the Entries enumerator.

```csharp
using (var archive = RarArchive.Open(fileStream))
{
    //foreach (var entry in archive.Entries)
    var entries = archive.Entries.GetEnumerator();
    while (entries.MoveNext()) // <-- SharpCompress.Common.InvalidFormatException: 'Unknown Rar Header: 205'
...
}
```

<img width="533" alt="image" src="https://user-images.githubusercontent.com/1046558/163496146-9852cba1-9d37-4626-9d60-3e347661a9b0.png">

**Comment by adamhathcock** [comment]
SharpComrpess is more raw than WinRAR.  You'll have to manually try/catch errors like that.

If there's a better way to handle corrupt entries and skip them, I'm open to changing the API.

**Comment by pgodwin** [comment]
Strangely, if I call to `RarArchive.Entries.Count`, before looping through the entries, the exception isn't raised.

**Comment by adamhathcock** [comment]
The entries are built only using the header that has file info then skips to the next header without reading everything.  

It's likely that a file is stored/compressed incorrectly but the recorded sizes are fine.

**Comment by pgodwin** [comment]
I think thats it. I tried creating some purposefully corrupted RAR files to try and reproduce the issue, but couldn't reproduce the issue (only ended up creating invalid vints or changing the number of bytes in the file). 

I'm happy to close this issue as it's clearly by design and isn't a bug in the code. 

At this point, I don't have any API suggestions to improve things - it'd be good to know if an entry is invalid or can't be read, and be able to skip to the next item - but how you would do so cleanly I'm not too sure (perhaps overload `IEnumerator.MoveNext()` with an option to ignore bad entries / add an State/Exception field to the collection).

**Comment by adamhathcock** [comment]
Yeah, the problem is that you can only "detect" it by trying to decompress the file.  Putting a try/catch in the correct place is the trick but doing it in a clean way is the hard part.
